sites/all/modules/hid/hid_profiles/hid_profiles.pages.inc

```diff
@@ -67,71 +79,70 @@ function hid_profiles_settings_form($form, &$form_state) {
  * @param boolean $extended
  *  Option to return extended information about organizations.
  *
- * @return
- *  JSON output of the matching Organizations array.
+ * @TODO: Convert this to an endpoint for HID.
  */
 function hid_organizations_autocomplete($org_names, $extended = FALSE) {
   $matches = array();
-
-  // Get the vocabulary for the Organizations to limit the term query by VID.
-  $vocabulary = taxonomy_vocabulary_machine_name_load('hr_organization');
+  if (empty($org_names)) {
+    _hid_profiles_json_helper($matches);
+  }
 
   // Determine the max number of Organizations to return per request.
   $limit = variable_get('hid_organizations_autocomplete_limit', 30);
-
-  // Limit the query to "begins with" or "contains"
+  // Limit the query to "starts with" or "contains"
   $contains = variable_get('hid_organizations_autocomplete_true', TRUE);
-  $contains_delimiter = ($contains) ? '%' : '';
-
-  // Search for hr_organization taxonomy terms, where the term name or
-  // organization field value match the given search string.
-  $query = db_select('taxonomy_term_data', 't');
-  $query->leftJoin('field_data_field_acronym', 'fdfa', '(t.tid = fdfa.entity_id AND fdfa.entity_type = :fdfa_type)', array(':fdfa_type' => 'taxonomy_term'));
-  $query->leftJoin('field_data_field_organization_type', 'fot', 't.tid = fot.entity_id');
-  $query->leftJoin('field_data_name_field', 'fnf', 'fot.field_organization_type_target_id = fnf.entity_id');
-
-  $query
-    ->fields('t')
-    ->fields('fdfa')
-    ->fields('fot')
-    ->fields('fnf')
-    ->condition('t.vid', $vocabulary->vid)
-    ->condition('fnf.language', 'en')
-    ->condition('fdfa.language', 'en');
-
-  $or = db_or();
-  $or->condition('t.name', $contains_delimiter . db_like($org_names) . '%', 'LIKE');
-  $or->condition('fdfa.field_acronym_value', $contains_delimiter . db_like($org_names) . '%', 'LIKE');
-
-  $return = $query
-    ->condition($or)
+  $operator = ($contains) ? 'CONTAINS' : 'STARTS_WITH';
+
+  // Query the terms by name.
+  $query = new EntityFieldQuery();
+  $results = $query
+    ->entityCondition('entity_type', 'taxonomy_term')
+    ->entityCondition('bundle', 'hr_organization')
+    ->propertyCondition('name', db_like($org_names), $operator)
     ->range(0, $limit)
     ->execute();
+  $data = reset($results);
+
+  // If there are not enough results, look for acronyms as well.
+  if (count($data) < $limit) {
+    $query = new EntityFieldQuery();
+    $results = $query
+      ->entityCondition('entity_type', 'taxonomy_term')
+      ->entityCondition('bundle', 'hr_organization')
+      ->fieldCondition('field_acronym', 'value', db_like($org_names), $operator)
+      ->propertyCondition('tid', array_keys($data), 'NOT IN')
+      ->range(0, $limit)
+      ->execute();
+    $data += reset($results);
```

Comment:
@pcambra I'm getting an "unsupported operand types" on this line. Did you test it in your env ? If so and if it works, which version of PHP are you running ?